File: src/ClashBoard.Api/Models/Entities/Player.cs
namespace ClashBoard.Api.Models.Entities;

public class Player
{
    public Guid Id { get; set; }
    public required string GamerTag { get; set; }
    public string? RealName { get; set; }
    public string? MainCharacter { get; set; }
    public string? SecondaryCharacter { get; set; }
    public string? AvatarUrl { get; set; }
    public string? Region { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    // Navigation
    public ICollection<LeaguePlayer> LeaguePlayers { get; set; } = [];
    public ICollection<Match> MatchesAsPlayer1 { get; set; } = [];
    public ICollection<Match> MatchesAsPlayer2 { get; set; } = [];
}

File: client/src/types/player.ts
export interface PlayerLeague {
  id: string;
  leagueId: string;
  playerId: string;
  wins: number;
  losses: number;
  draws: number;
  points: number;
  matchesPlayed: number;
  gamesWon: number;
  gamesLost: number;
  currentWinStreak: number;
  bestWinStreak: number;
  recentForm: string | null;
  joinedAt: string;
  league: {
    id: string;
    name: string;
  };
}

export interface Player {
  id: string;
  gamerTag: string;
  realName: string | null;
  mainCharacter: string | null;
  secondaryCharacter: string | null;
  avatarUrl: string | null;
  region: string | null;
  createdAt: string;
  updatedAt: string | null;
  leaguePlayers: PlayerLeague[];
}
